sources/api/early-boot-config/src/main.rs

```diff
@@ -454,3 +195,35 @@ async fn main() {
         process::exit(1);
     }
 }
+
+mod error {
+    use http::StatusCode;
+    use snafu::Snafu;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility = "pub(super)")]
+    pub(super) enum Error {
```

Comment:
We got together and implemented this to see what it would look like. It looks like this:

https://github.com/zmrow/bottlerocket/compare/ebc_refactor...webern:ebc_error

There's some cleanup, but the basic premise is that one doesn't not have to introduce any conditional compilation into the provider's error type. This would keep the surface area of conditional compilation restricted to one place, (the instantiation of the provider).

Reply:
Addressed in my comments below!